src/Template/Element/advanced_search_form.ctp

```diff
@@ -44,6 +44,7 @@ echo $this->Form->create('AdvancedSearch', [
                     'dir' => 'auto',
                 ));
                 ?>
+                <md-icon id="clearSearch" tabindex="-1" ng-click="ctrl.clearSearch()">clear</md-icon>
```

Comment:
Assuming you've added the directive that I posted in my other comment, you can replace this with:
```
<md-icon reset-button>clear</md-icon>
```

Reply:
I tried this code, I still have to implement the javascript for the actual reset, but I don0t understand why in search bar the `md-icon` acts as a button while if I add it in advanced search page, it is a simple icon.
To make the button work I had to put the `md-icon` inside and `md-button`. Then it works and the directive is called correctly, only problem is that then the button is too large.

Reply:
It does already work as a button. But your expression for `ng-click` does nothing because `ctrl.clearSearch` is not defined.

AngularJS makes quick testing a little bit harder because some features aren't everywhere available (such as for example showing a quick pop-up with `alert()`).

But if you add `<?= "{{test}}" ?>` just before the line with `md-icon` and change the value of `ng-click` to `"test = test + 1"` you should get a simple counter which increments whenever you click the icon.

Reply:
I was following Trang suggestion to use a directive and it actually works when called, but that does not work with the plain `md-icon` and requires the `md-button`. For now I will focus on te actual JS to reset and then check what I can do with styling.

Reply:
Hmm... The directive should be called even if you don't put the `md-icon` inside a `md-button`. Perhaps you were confused at the fact that on hover, the cursor isn't converted into a hand cursor?

In the search bar, the icon acts as a button because the CSS for this element contains `cursor: pointer;`
```
.search_bar #clearSearch {
    background: transparent;
    border: none;
    cursor: pointer;
    color: #CCCCCC;
    margin-left: -30px;
}
```

Reply:
Probably I was confused by the hovering, I checked again on another reset button in the page and the click works even in that. Probably yesterday I messed to much with code and at beginning it didn't work JS and I trid other solutions. Anyway I go on with JS and then I will check the style in CSS.

Reply:
I have updated and pushed. Javascript code works, I don't know it it is the most elegant solution as it heavily depends on order of html elements or if there's a better way.
For css I can edit `advanced_search.css` right?

Reply:
For the CSS, you should rather edit the `default.css`. I've experimented a bit and I think you could use this code:

```
.md-button-right input {
    padding-right: 40px;
}
.md-button-right .md-icon-button {
    position: absolute;
    right: -8px;
    top: -4px;
}
```

You can add it at the end of the `default.css` in the "Angular Material override" section.

With this, you can to set `class="md-button-right"` (instead of `md-icon-right`) on the `<md-input-container>`, and you can encapsulate the `<md-icon>` inside a `<md-button class="md-icon-button">`. It does have the advantage to provide an extra little animation on click, which the `<md-icon>` alone wouldn't have.

For the Javascript, how about adding an attribute to specify the id of the target input, then read the value of this attribute to reset the input which has that id?
```
<input id="some-id"/>
<md-button reset-button target="some-id"> ...
```